File: Practica_1/src/gestorAplicacion/noPersonas/OrdenPresencial.java
package gestorAplicacion.noPersonas;
import java.util.ArrayList;
import gestorAplicacion.personas.*;
import uiMain.ActivarPromocion;


public class OrdenPresencial implements Orden {
	
	
	private static int numeroOrdenes = 1000;
	private ArrayList<Bebida> bebidas = new ArrayList<Bebida> ();
	private Cliente cliente;
	private Barista barista;
	private int numeroOrden;
	private float costo;
	private static int ordenesCreadas;
	
	
	public OrdenPresencial(ArrayList<Bebida> bebidas, Cliente cliente, Barista barista) {
		this();
		this.numeroOrden = numeroOrdenes;
		numeroOrdenes ++;
		this.bebidas = bebidas;
		this.cliente = cliente;
		this.barista = barista;
		float acumulado = 0;
		for (Bebida x: bebidas) acumulado += x.getPrecio();
		this.costo = acumulado;
	}
	
	public OrdenPresencial() {
		ordenesCreadas++;
	}
	
	public ArrayList<Bebida> getBebidas() {
		return bebidas;
	}

	public static int getOrdenesCreadas() {
		return ordenesCreadas;
	}

	public void setBebidas(ArrayList<Bebida> bebidas) {
		this.bebidas = bebidas;
	}

	public static void setOrdenesCreadas(int ordenesCreadas) {
		OrdenPresencial.ordenesCreadas = ordenesCreadas;
	}

	public void aplicarPromocion(Promocion promocion) {
		if(promocion.equals(Promocion.CAFE2X1) && ActivarPromocion.isPromos()) {
			this.costo -= 5000;
		} else if(promocion.equals(Promocion.JUGO2X1) && ActivarPromocion.isPromos()) {
			this.costo -= 4000;
		} else if (promocion.equals(Promocion.CAFECONJUGOAL50) && ActivarPromocion.isPromos()){
			this.costo -= 4500;
		}
	}
	
	public String toString() {
		return this.barista.getNombre() + " le vendió a " + this.cliente.getNombre()+" la orden número " + this.numeroOrden + " por un valor de " + this.costo;
	}


	public static int getNumeroOrdenes() {
		return numeroOrdenes;
	}


	public ArrayList<Bebida> getProductos() {
		return bebidas;
	}


	public Cliente getCliente() {
		return cliente;
	}


	public Barista getBarista() {
		return barista;
	}


	public int getNumeroOrden() {
		return numeroOrden;
	}


	public float getCosto() {
		return costo;
	}


	public static void setNumeroOrdenes(int numeroOrdenes) {
		OrdenPresencial.numeroOrdenes = numeroOrdenes;
	}


	public void setProductos(ArrayList<Bebida> bebidas) {
		this.bebidas = bebidas;
	}


	public void setCliente(Cliente cliente) {
		this.cliente = cliente;
	}


	public void setBarista(Barista barista) {
		this.barista = barista;
	}


	public void setNumeroOrden(int numeroOrden) {
		this.numeroOrden = numeroOrden;
	}


	public void setCosto(float costo) {
		this.costo = costo;
	}
	
}

File: Practica_2/noPersonas/ordenPresencial.py
from noPersonas.orden import Orden
from personas.barista import Barista
#from personas.cliente import Cliente
from noPersonas.promocion import Promocion

class OrdenPresencial(Orden): 
    _numeroOrdenes = 1000;
    _ordenesCreadas = 0
    
    def __init__(self, bebidas, cliente, barista):
        OrdenPresencial._ordenesCreadas += 1
        self._numeroOrden = OrdenPresencial._numeroOrdenes
        OrdenPresencial._numeroOrdenes += 1
        self._bebidas = bebidas
        self._cliente = cliente
        self._barista = barista
        acumulado = 0;
        barista._ventas.append(self)
		
        for x in self._bebidas:
            acumulado += x.getPrecio()
            
        self._costo = acumulado
    
    def getBebidas(self):
        return self._bebidas;
    
    @classmethod
    def getOrdenesCreadas(cls):
        return cls._ordenesCreadas
    
    def setBebidas(self, bebidas):
        self._bebidas = bebidas

    @classmethod
    def setOrdenesCreadas(cls, ordenesCreadas):
        cls._ordenesCreadas = ordenesCreadas

    def aplicarPromocion(self, promocion):
        if promocion == Promocion.CAFE2X1: # && ActivarPromocion.isPromos():
            self._costo -= 5000
        elif promocion == Promocion.JUGO2X1: # && ActivarPromocion.isPromos():
            self._costo -= 4000
        elif promocion == Promocion.CAFECONJUGOAL50: # && ActivarPromocion.isPromos()):
            self._costo -= 4500
    
    def __str__(self):
        return self._barista.getNombre() + ' le vendió a ' + self._cliente.getNombre()+' la orden número ' + str(self._numeroOrden) + ' por un valor de ' + str(self._costo)

    @classmethod
    def getNumeroOrdenes(cls):
        return cls._numeroOrdenes
    
    def getProductos(self):
        return self._bebidas
    
    def getCliente(self):
        return self._cliente
    
    def getBarista(self):
        self._barista
    
    def getNumeroOrden(self):
        return self._numeroOrden
    
    def getCosto(self):
        return self._costo

    @classmethod
    def setNumeroOrdenes(cls, numeroOrdenes):
        cls._numeroOrdenes = numeroOrdenes;
    
    def setProductos(self, bebidas):
        self._bebidas = bebidas
    
    def setCliente(self, cliente):
        self._cliente = cliente

    def setBarista(self, barista):
        self._barista = barista

    def setNumeroOrden(self, numeroOrden):
        self.numeroOrden = numeroOrden
    
    def setCosto(self, costo):
        self._costo = costo